### PR TITLE
Strip leading and trailing spaces in redirects

### PR DESCRIPTION
**Pull request by chsamlee**

### CODE REVIEW

**Comment by michaelsilver** [comment]
Could you add a unit test for the article that caused the code to trip up before you added the one line to strip whitespace?

File: tests/test_util.py [hunk 1]
```diff
@@ -202,6 +202,11 @@ def test_redirect_with_leading_colon(self):
         self.assertEquals('Jelgava–Liepāja Railway',
                           util.extract_redirect(markup))
 
+    def test_redirect_with_trailing_space(self):
+        markup = "#REDIRECT [[Everybody's Got to Learn Sometime ]]"
```
Comment (michaelsilver) [inline tests/test_util.py]: Because "Everything in Wikipedia is hard" ™️ ©️ ®️ and we want to be robust,  why don't you add a space in front of the article, in case someone does that somewhere on Wikipedia. Or, do you think that is illegal markup, so it would never happen?

Comment (chsamlee) [inline tests/test_util.py]: I think that is illegal markup, but I can't find a source stating that. But then I can't find any rule stating that leading and trailing spaces are allowed either.

Comment (michaelsilver) [inline tests/test_util.py]: Why not just add a leading space, update the unit test name accordingly, and then move on comforted that you'll probably never encounter this bug again, even if a leading space might never actually happen IRL?

**Comment by michaelsilver** [comment]
LGTM!
